src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,83 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_user_isOK() {
+        User expected = new User("BobGood", "1234567", 18);
+        User actual = registrationService.register(expected);
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void register_userPasswordIsShort_isNotOK() {
+        User expected = new User("BobWithShortPassword", "12", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userLoginIsShort_isNotOK() {
+        User expected = new User("Bob", "1234567", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userUnderage_isNotOK() {
+        User expected = new User("BobChild", "1234567", 15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
```

Comment:
You are testing the case where the user is underage, which is good. However, you should also test the edge case where the user is exactly at the minimum age. This will help ensure that your code correctly handles this edge case.